langchain4j-alloydb-pg/src/main/java/dev/langchain4j/store/embedding/alloydb/AlloyDBEmbeddingStore.java

```diff
@@ -80,32 +80,30 @@ public class AlloyDBEmbeddingStore implements EmbeddingStore<TextSegment> {
      * @param queryOptions (Optional) QueryOptions class with vector search
      * parameters
      */
-    public AlloyDBEmbeddingStore(String tableName, String schemaName, String contentColumn, String embeddingColumn,
-            String idColumn, List<MetadataColumn> metadataColumns, String metadataJsonColumn, List<String> ignoreMetadataColumnNames,
-            DistanceStrategy distanceStrategy, Integer k, Integer fetchK, Double lambdaMult, QueryOptions queryOptions, AlloyDBEngine engine) {
-        this.engine = engine;
-        this.tableName = tableName;
-        this.schemaName = schemaName;
-        this.contentColumn = contentColumn;
-        this.embeddingColumn = embeddingColumn;
-        this.idColumn = idColumn;
-        this.metadataJsonColumn = metadataJsonColumn;
-        this.metadataColumns = metadataColumns;
-        this.distanceStrategy = distanceStrategy;
-        this.k = k;
-        this.fetchK = fetchK;
-        this.lambdaMult = lambdaMult;
-        this.queryOptions = queryOptions;
+    public AlloyDBEmbeddingStore(Builder builder) {
 
         // check columns exist in the table
-        verifyEmbeddingStoreColumns(ignoreMetadataColumnNames);
+        verifyEmbeddingStoreColumns(builder.getIgnoreMetadataColumnNames());
+
+        this.engine = builder.getEngine();
+        this.tableName = builder.getTableName();
+        this.schemaName = builder.getSchemaName();
+        this.contentColumn = builder.getContentColumn();
+        this.embeddingColumn = builder.getEmbeddingColumn();
+        this.idColumn = builder.getIdColumn();
+        this.metadataJsonColumn = builder.getMetadataJsonColumn();
+        this.metadataColumns = builder.getMetadataColumns();
+        this.distanceStrategy = builder.getDistanceStrategy();
+        this.k = builder.getK();
+        this.fetchK = builder.getFetchK();
+        this.lambdaMult = builder.getLambdaMult();
+        this.queryOptions = builder.getQueryOptions();
```

Comment:
Sorry, you were correct on this:
```suggestion
        this.engine = builder.getEngine();
        this.tableName = builder.getTableName();
        this.schemaName = builder.getSchemaName();
        this.contentColumn = builder.getContentColumn();
        this.embeddingColumn = builder.getEmbeddingColumn();
        this.idColumn = builder.getIdColumn();
        this.metadataJsonColumn = builder.getMetadataJsonColumn();
        this.metadataColumns = builder.getMetadataColumns();
        this.distanceStrategy = builder.getDistanceStrategy();
        this.k = builder.getK();
        this.fetchK = builder.getFetchK();
        this.lambdaMult = builder.getLambdaMult();
        this.queryOptions = builder.getQueryOptions();
        verifyEmbeddingStoreColumns(builder.getIgnoreMetadataColumnNames());
```

Reply:
no problemo